### PR TITLE
feat(thermocycler-gen2): thermistor drift check

### PR DESCRIPTION
**Pull request by fsinapi**
### Summary

Adds the thermistor drift error from TC Gen1. Similar to the Gen1 firmware, this error is persistent until the device resets. A "clear error" command may be appropriate in a future PR.

This has not been tested on hardware yet, and should not be merged until that testing can occur.

EDIT: Tested on hardware by unplugging a thermistor after reaching temperature. After doing this, any attempt to set plate temperature returned an `ERR408` message.

### CODE REVIEW

File: stm32-modules/include/thermocycler-gen2/thermocycler-gen2/plate_control.hpp [hunk 1]
```diff
@@ -159,6 +163,21 @@ class PlateControl {
      */
     [[nodiscard]] auto temp_within_setpoint() const -> bool;
 
+    /**
+     * @brief Check for thermistor drift.
+     * @return true if the thermistors are \b within spec, false if the
```
Comment (sfoster1) [inline stm32-modules/include/thermocycler-gen2/thermocycler-gen2/plate_control.hpp:168]: `\b` ?

**Review by sfoster1 — APPROVED** [review]
Looks good to me, though like you say we should test it.

Another thing to think about - should this be latching until power cycle? I know that's what the gen1 does but that doesn't mean it's the right thing.

File: stm32-modules/include/thermocycler-gen2/thermocycler-gen2/plate_control.hpp (hunk 1 above)
Comment (fsinapi) [reply stm32-modules/include/thermocycler-gen2/thermocycler-gen2/plate_control.hpp:168]: I like using [Doxygen formatting](https://www.doxygen.nl/manual/commands.html#cmdb)

**Comment by fsinapi** [comment]
> Looks good to me, though like you say we should test it.
> 
> Another thing to think about - should this be latching until power cycle? I know that's what the gen1 does but that doesn't mean it's the right thing.

Per outside discussion, the plan will be to add a command that will clear this error.

**Pull request merged**